File: packages/ash-dotnet/AshErrorCode.cs
namespace Ash;

/// <summary>
/// ASH error codes.
/// </summary>
public enum AshErrorCode
{
    InvalidContext,
    ContextExpired,
    ReplayDetected,
    IntegrityFailed,
    EndpointMismatch,
    ModeViolation,
    UnsupportedContentType,
    MalformedRequest,
    CanonicalizationFailed
}

/// <summary>
/// Extension methods for AshErrorCode.
/// </summary>
public static class AshErrorCodeExtensions
{
    /// <summary>
    /// Convert error code to string representation.
    /// </summary>
    public static string ToErrorString(this AshErrorCode code) => code switch
    {
        AshErrorCode.InvalidContext => "INVALID_CONTEXT",
        AshErrorCode.ContextExpired => "CONTEXT_EXPIRED",
        AshErrorCode.ReplayDetected => "REPLAY_DETECTED",
        AshErrorCode.IntegrityFailed => "INTEGRITY_FAILED",
        AshErrorCode.EndpointMismatch => "ENDPOINT_MISMATCH",
        AshErrorCode.ModeViolation => "MODE_VIOLATION",
        AshErrorCode.UnsupportedContentType => "UNSUPPORTED_CONTENT_TYPE",
        AshErrorCode.MalformedRequest => "MALFORMED_REQUEST",
        AshErrorCode.CanonicalizationFailed => "CANONICALIZATION_FAILED",
        _ => "UNKNOWN_ERROR"
    };
}

File: packages/ash-php/src/AshErrorCode.php
<?php

declare(strict_types=1);

namespace Ash;

/**
 * ASH error codes.
 */
enum AshErrorCode: string
{
    case InvalidContext = 'INVALID_CONTEXT';
    case ContextExpired = 'CONTEXT_EXPIRED';
    case ReplayDetected = 'REPLAY_DETECTED';
    case IntegrityFailed = 'INTEGRITY_FAILED';
    case EndpointMismatch = 'ENDPOINT_MISMATCH';
    case ModeViolation = 'MODE_VIOLATION';
    case UnsupportedContentType = 'UNSUPPORTED_CONTENT_TYPE';
    case MalformedRequest = 'MALFORMED_REQUEST';
    case CanonicalizationFailed = 'CANONICALIZATION_FAILED';
}
